openedx/core/djangoapps/course_groups/views.py

```diff
@@ -145,31 +150,35 @@ def course_cohort_settings_handler(request, course_key_string):
             settings_to_change['is_cohorted'] = request.json.get('is_cohorted')
 
         if 'cohorted_course_wide_discussions' in request.json or 'cohorted_inline_discussions' in request.json:
-            cohorted_course_wide_discussions = request.json.get(
-                'cohorted_course_wide_discussions', cohorted_course_wide_discussions
+            divided_course_wide_discussions = request.json.get(
+                'cohorted_course_wide_discussions', divided_course_wide_discussions
             )
-            cohorted_inline_discussions = request.json.get(
-                'cohorted_inline_discussions', cohorted_inline_discussions
+            divided_inline_discussions = request.json.get(
+                'cohorted_inline_discussions', divided_inline_discussions
             )
-            settings_to_change['cohorted_discussions'] = cohorted_course_wide_discussions + cohorted_inline_discussions
+            settings_to_change['divided_discussions'] = divided_course_wide_discussions + divided_inline_discussions
 
         if 'always_cohort_inline_discussions' in request.json:
-            settings_to_change['always_cohort_inline_discussions'] = request.json.get(
+            settings_to_change['always_divide_inline_discussions'] = request.json.get(
                 'always_cohort_inline_discussions'
             )
 
         if not settings_to_change:
             return JsonResponse({"error": unicode("Bad Request")}, 400)
 
         try:
-            cohort_settings = cohorts.set_course_cohort_settings(
-                course_key, **settings_to_change
-            )
+            if 'is_cohorted' in settings_to_change:
+                is_cohorted = settings_to_change['is_cohorted']
+                cohorts.set_course_cohorted(course_key, is_cohorted)
+                del settings_to_change['is_cohorted']
+                settings_to_change['division_scheme'] = DIVIDED_BY_COHORT if is_cohorted else NOT_DIVIDED
```

Comment:
Note to reviewers: I was missing line 174, and all tests passed. However, enabling cohorts on a course would not cause discussions to be cohorted (because division_scheme was "none"). We need to add test coverage for this.